pkg/deployment/config_test.go

```diff
@@ -334,6 +336,89 @@ func TestControllerConfiguration(t *testing.T) {
 			DefaultAffinityType:                 defaultAffinityTypeValue,
 		},
+	}, {
+		name:    "runtime class name defaults to nothing",
+		wantErr: false,
+		data: map[string]string{
+			QueueSidecarImageKey: defaultSidecarImage,
+		},
+		wantConfig: &Config{
+			DigestResolutionTimeout:        digestResolutionTimeoutDefault,
+			ProgressDeadline:               ProgressDeadlineDefault,
+			QueueSidecarCPURequest:         &QueueSidecarCPURequestDefault,
+			QueueSidecarImage:              defaultSidecarImage,
+			QueueSidecarTokenAudiences:     sets.New(""),
+			RegistriesSkippingTagResolving: sets.New("kind.local", "ko.local", "dev.local"),
+			RuntimeClassNames:              nil,
+			DefaultAffinityType:            defaultAffinityTypeValue,
+		},
+	}, {
+		name:    "runtime class name with wildcard",
+		wantErr: false,
+		wantConfig: &Config{
+			RuntimeClassNames: map[string]RuntimeClassNameLabelSelector{
+				"gvisor": RuntimeClassNameLabelSelector{},
+			},
+			DigestResolutionTimeout:        digestResolutionTimeoutDefault,
+			ProgressDeadline:               ProgressDeadlineDefault,
+			QueueSidecarCPURequest:         &QueueSidecarCPURequestDefault,
+			QueueSidecarImage:              defaultSidecarImage,
+			QueueSidecarTokenAudiences:     sets.New(""),
+			RegistriesSkippingTagResolving: sets.New("kind.local", "ko.local", "dev.local"),
+			DefaultAffinityType:            defaultAffinityTypeValue,
+		},
+		data: map[string]string{
+			"runtime-class-name": "gvisor: {}",
+			QueueSidecarImageKey: defaultSidecarImage,
+		},
+	}, {
+		name:    "runtime class name with wildcard and label selectors",
+		wantErr: false,
+		wantConfig: &Config{
+			RuntimeClassNames: map[string]RuntimeClassNameLabelSelector{
+				"gvisor": RuntimeClassNameLabelSelector{},
+				"kata": RuntimeClassNameLabelSelector{
+					Selector: map[string]string{
+						"some": "value-here",
+					},
+				},
+			},
+			DigestResolutionTimeout:        digestResolutionTimeoutDefault,
+			ProgressDeadline:               ProgressDeadlineDefault,
+			QueueSidecarCPURequest:         &QueueSidecarCPURequestDefault,
+			QueueSidecarImage:              defaultSidecarImage,
+			QueueSidecarTokenAudiences:     sets.New(""),
+			RegistriesSkippingTagResolving: sets.New("kind.local", "ko.local", "dev.local"),
+			DefaultAffinityType:            defaultAffinityTypeValue,
+		},
+		data: map[string]string{
+			"runtime-class-name": `---
+gvisor: {}
+kata:
+  selector:
+    some: value-here
+`,
+			QueueSidecarImageKey: defaultSidecarImage,
+		},
+	}, {
+		name:    "runtime class name with bad label selectors",
+		wantErr: true,
+		data: map[string]string{
+			QueueSidecarImageKey: defaultSidecarImage,
+			"runtime-class-name": `---
+gvisor: {}
+kata:
+  selector:
+    "-a": " a  a "
+`,
+		},
+	}, {
+		name:    "runtime class name with an unparsable format",
+		wantErr: true,
+		data: map[string]string{
+			QueueSidecarImageKey: defaultSidecarImage,
+			"runtime-class-name": ` ???; 231424 `,
```

Comment:
To be clear the error here is captured due to yaml unmarshalling: ```error unmarshaling JSON: while decoding JSON: json: cannot unmarshal string into Go value of type map[string]deployment.RuntimeClassNameLabelSelector
```

Reply:
I would like to implement the use of ValidateRuntimeClassName as suggested.
If the YAML cannot be parsed, how should the validation be handled?

Reply:
I was trying to emphasize that some K8s validation might be needed. After yaml is parsed we can check for a valid name according to K8s.

Reply:
Agree, it makes a lot of sense to have it.